collector/infiniband_linux_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestDevices(t *testing.T) {
+	devices, err := infinibandDevices("fixtures/sys/class/infiniband", errors.New("InfiniBand / No devices detected"))
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(devices) != 1 {
+		t.Fatal(err)
```

Comment:
This will just do `t.Fatal(nil)` in effect.  Should add a more descriptive error message.